apps/web/src/pages/templates/editor_v2/useWorkflowDetailPageForm.ts

```diff
@@ -30,34 +36,44 @@ export const useWorkflowDetailPageForm = ({ templateId, workflow }: UseWorkflowD
     },
   });
 
-  const hasChanges = Object.keys(formState.dirtyFields).length > 0;
+  const hasChanges = Object.keys(dirtyFields).length > 0;
 
-  const onSubmit: SubmitHandler<WorkflowDetailFormContext> = () => {
-    if (formState.dirtyFields?.preferences) {
-      updateWorkflowChannelPreferences(getValues('preferences'));
-    }
+  const onSubmit: SubmitHandler<WorkflowDetailFormContext> = async (data) => {
+    try {
+      if (dirtyFields?.preferences) {
+        updateWorkflowChannelPreferences(getValues('preferences'));
+      }
 
-    if (formState.dirtyFields?.general) {
-      const { workflowId, ...templateValues } = getValues('general');
-      updateTemplateMutation({ id: templateId, data: { ...templateValues, identifier: workflowId } });
-    }
+      if (dirtyFields?.general) {
+        const { workflowId, ...templateValues } = getValues('general');
+        await updateTemplateMutation({ id: templateId, data: { ...templateValues, identifier: workflowId } });
+      }
 
-    reset();
+      successMessage('Workflow updated successfully');
+
+      reset(data);
+    } catch (e: any) {
+      errorMessage(e.message || 'Unexpected error occurred');
+    }
   };
 
-  useEffect(() => {
-    if (workflow) {
-      setValue('general', {
+  useEffectOnce(() => {
+    if (!workflow) return;
+
+    const form = {
+      general: {
         workflowId: workflow.triggers?.[0]?.identifier ?? '',
         name: workflow.name,
         description: workflow.description,
-      });
-    }
-  }, [workflow]);
+      },
+    };
+    reset(form);
+  }, !!workflow);
```

Comment:
For the initial values

Reply:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
💭 **thought (non-blocking):** The reason this was intended to run every time `workflow` changes is in the chance the back-end does some sort of transformation on the data, we'd want that to be the source of truth rather than letting the form hold the state.‏